pootle/apps/pootle_app/models/directory.py

```diff
@@ -154,15 +154,17 @@ def can_be_updated(self):
     def get_children(self):
         result = []
         if not self.is_projects_root():
-            # FIXME: can we replace this with a quicker path query?
-            result.extend([item for item in
-                           self.child_stores.live().iterator()])
-            result.extend([item for item in self.child_dirs.live().iterator()])
+            result.extend(
+                self.child_stores.live().select_related(
+                    "parent", "translation_project"))
+            result.extend(self.child_dirs.live())
```

Comment:
@phlax, I'm not sure about this. You call this "optimization" but I can reckon that parents are not necessary when we calculate stats. Am I wrong here?
Can you tell me when we need parent objects for children?